cmd/tracee/main.go

```diff
@@ -0,0 +1,258 @@
+package main
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	"github.com/aquasecurity/tracee/cmd/tracee-ebpf/flags"
+	"github.com/aquasecurity/tracee/pkg/cmd"
+	"github.com/aquasecurity/tracee/pkg/events"
+	"github.com/aquasecurity/tracee/pkg/logger"
+	"github.com/aquasecurity/tracee/pkg/rules/engine"
+	"github.com/aquasecurity/tracee/pkg/rules/signature"
+	"github.com/aquasecurity/tracee/pkg/server"
+	"github.com/aquasecurity/tracee/types/detect"
+
+	cli "github.com/urfave/cli/v2"
+)
+
+func init() {
+	// Avoiding to override package-level logger
+	// when it's already set by logger environment variables
+	if !logger.IsSetFromEnv() {
+		// Logger Setup
+		logger.Init(
+			&logger.LoggerConfig{
+				Writer:    os.Stderr,
+				Level:     logger.InfoLevel,
+				Encoder:   logger.NewJSONEncoder(logger.NewProductionConfig().EncoderConfig),
+				Aggregate: false,
+			},
+		)
+	}
+}
+
+var version string
+
+func main() {
+	app := &cli.App{
+		Name:    "Tracee",
+		Usage:   "Trace OS events and syscalls using eBPF",
+		Version: version,
+		Action: func(c *cli.Context) error {
+			ctx := context.Background()
+
+			if c.NArg() > 0 {
+				return cli.ShowAppHelp(c) // no args, only flags supported
+			}
+
+			flags.PrintAndExitIfHelp(c)
+
+			// Rego command line flags
+
+			rego, err := flags.PrepareRego(c.StringSlice("rego"))
+			if err != nil {
+				return err
+			}
+
+			// event names are parsed here because we need to know which signatures to load
+			eventNames := getEventNames(c)
+
+			sigs, err := signature.Find(
+				rego.RuntimeTarget,
+				rego.PartialEval,
+				c.String("rules-dir"),
+				eventNames,
+				rego.AIO,
+			)
+
+			if err != nil {
+				return err
+			}
+
+			createEventsFromSignatures(sigs)
+
+			if c.Bool("list") {
+				cmd.PrintEventList(true) // list events
+				return nil
+			}
+
+			traceeConfig, printerConfig, err := cmd.GetTraceeAndPrinterConfig(c, version)
+			if err != nil {
+				return err
+			}
+
+			// parse arguments must be enabled if the rule engine is part of the pipeline
+			traceeConfig.Output.ParseArguments = true
+
+			traceeConfig.EngineConfig = engine.Config{
+				Enabled:             true,
+				Signatures:          sigs,
+				SignatureBufferSize: c.Uint("sig-buffer"),
+			}
+
+			return cmd.Run(ctx, c, traceeConfig, printerConfig)
+		},
+		Flags: []cli.Flag{
```

Comment:
Also considered this refactor, and opted not to change now, because if we decide to drop the previous experience to have only `cmd/tracee` this won't be necessary either way, the PR was getting to big and this refactor along a couple of others for the `cmd/tracee-ebpg/*` was purposely not done.

Reply:
I think we shouldn't mess with the old tracee-ebpf too much in the first PRs. New things must reside only here. This will facilitate parallel PRs from other tasks.